cangen.c

```diff
@@ -398,6 +405,32 @@ enum {
 	OPT_START = UCHAR_MAX + 2,
 };
 
+/*
+ * Search for '*' in dataoptarg, save its position, replace it with 0.
```

Comment:
I'm not sure if it's a good idea to have an asterisk `*`  here. Can this interfere with the expansion of `*` in shell scripts?

Would `x` also be ok for you?

Reply:
Not sure - any shell expert here? However `*` is usable for example in nmap so I didn't expect any issues and it sounds the most user-friendly for me.

If you want, we can change to `x`.

Additionaly, the formatting character shouldn't be hardcoded, maybe some #define is better?

Reply:
> Not sure - any shell expert here? However `*` is usable for example in nmap so I didn't expect any issues and it sounds the most user-friendly for me.
> 
> If you want, we can change to `x`.
> 
> Additionaly, the formatting character shouldn't be hardcoded, maybe some #define is better?

Yes, some define would be good.

AFAIK the shell expands the asterisk e.g. to a filename when it is placed in front or tail:
`AABB**DD` should be ok, but `AABBCCD*` would not be. Therefore `x` or maybe `_` seems safe.

Reply:
`*` changed to 'x', define is used

Reply:
The exact behavior of `*` depends on the shell (and probably configuration). For example on `zsh`:
```shell
echo AABB**DD 
zsh: no matches found: AABB**DD
```

on `bash`:

```shell
echo AABB**DD
AABB**DD
```

If the file exists, both shell will expand the `*`:

```shell
touch AABB--DD
echo AABB**DD
AABB--DD
```